### PR TITLE
feat(ci): automate production and beta releases to npm registry

### PR DESCRIPTION
**Pull request by vdovhanych**
This pr will allow us to publish releases to the npm registry with a simple button press. We will have 2 jobs one for production running only on v8 branch and beta one for all other branches with an option to upload beta tag packages.

### CODE REVIEW

**Review by szymonlesisz — APPROVED** [review]

**Comment by vdovhanych** [comment]
I have done some checks that @szymonlesisz  was asking for but I'm not really sure it's what he wanted. Check it if it's something you think we can work with and let me know. Yes, there is one small script in js that will check the local and remote versions, I think it's really not that great from me so I would appreciate any suggestions :)

File: scripts/ci-compare-version.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var compareVersions = require('compare-versions');
```
Comment (tomasklim) [inline scripts/ci-compare-version.js]:
```suggestion
const compareVersions = require('compare-versions');
```

File: .github/workflows/version_check.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    version_bump_check:
+      name: Check connect version bump
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+        - uses: actions/setup-node@v2
+        - run: |
+            npm install
+            npm i compare-versions
```
Comment (tomasklim) [inline .github/workflows/version_check.yml]: add to `package.json`

File: .github/workflows/version_check.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            [[ "$LAST_COMMIT" == "version $VERSION" ]]
+
+    version_bump_check:
+      name: Check connect version bump
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+        - uses: actions/setup-node@v2
+        - run: |
+            npm install
```
Comment (tomasklim) [inline .github/workflows/version_check.yml]: we use `yarn`

**Review by tomasklim — CHANGES_REQUESTED** [review]

**Review by tomasklim — APPROVED** [review]

**Comment by tsusanka** [comment]
@vdovhanych squash and force push? And let's merge

**Comment by tsusanka** [comment]
See here why the CI fails. https://github.com/trezor/connect/pull/848#issuecomment-883409848 It is not related to this PR.

**Review by tsusanka — APPROVED** [review]
LGTM, though @vdovhanych haven't we agreed we will prefer Gitlab over Github Actions? :) But let's leave it like this for now, it is OK

**Comment by vdovhanych** [comment]
Yes, @tsusanka I wanted to do it in Gitlab CI but it was not possible to do what we wanted. That is to run these checks on a pull requests to a v8 branch and as we do not push into the GitLab there is no other way. If we had these checks on Gitlab they would either run all the time which would mean that the pipeline would fail most of the time or we would have them run on push to a v8 branch but that doesn't provide us with preventing incorrect versioning in the v8 branch.

**Pull request merged**